src/audio/copier/copier.c

```diff
@@ -990,6 +987,49 @@ static int copier_get_hw_params(struct comp_dev *dev, struct sof_ipc_stream_para
 	return dai_common_get_hw_params(dd, dev, params, dir);
 }
 
+static int copier_bind(struct processing_module *mod, void *data)
+{
+	const struct ipc4_module_bind_unbind *const bu = (struct ipc4_module_bind_unbind *)data;
+	struct copier_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
+	struct comp_buffer *buffer;
+	uint32_t src_id, sink_id;
+	struct list_item *list;
+
+	src_id = IPC4_COMP_ID(bu->primary.r.module_id, bu->primary.r.instance_id);
+	sink_id = IPC4_COMP_ID(bu->extension.r.dst_module_id, bu->extension.r.dst_instance_id);
+
+	if (dev->ipc_config.id == src_id) {
+		/* update sink format */
+		list_for_item(list, &dev->bsink_list) {
+			buffer = container_of(list, struct comp_buffer, source_list);
+			uint32_t id = IPC4_SINK_QUEUE_ID(buf_get_id(buffer));
```

Comment:
There is a wrong macro used to retrieve sink id. Buffer id is created by `IPC4_COMP_ID(src_queue, dst_queue)`, in such a case `IPC4_SINK_QUEUE_ID` will return `dst_queue` id and not the `src_queue` id which we are interested in. Thus, you won't be able to update corresponding copier sink for sink num > 1.
`IPC4_SRC_QUEUE_ID()` should be used instead